c2rust-transpile/src/translator/assembly.rs

```diff
@@ -453,26 +453,26 @@ fn asm_is_att_syntax(asm: &str) -> bool {
     // Look for syntax directives.
     let intel_directive = asm.find(".intel_syntax");
     let att_directive = asm.find(".att_syntax");
-    if let (Some(intel_pos), Some(att_pos)) = (intel_directive, att_directive) {
-        // Both directives are present; presumably this asm switches to one at
-        // its start and restores the default at the end. Whichever comes first
-        // should be what the asm uses.
-        att_pos < intel_pos
-    } else if intel_directive.is_some() {
-        false
-    } else if att_directive.is_some() {
-        true
-    } else if asm.contains("word ptr") {
-        false
-    } else if asm.contains('$') || asm.contains('%') || asm.contains('(') {
-        // Guess based on sigils used in AT&T assembly:
-        // $ for constants, % for registers, and ( for address calculations
-        true
-    } else if asm.contains('[') {
-        false
-    } else {
-        // default to true, because AT&T is the default for gcc inline asm
-        true
+    match (intel_directive, att_directive) {
+        (Some(intel_pos), Some(att_pos)) => {
+            // Both directives are present; presumably this asm switches to one at
+            // its start and restores the default at the end. Whichever comes first
+            // should be what the asm uses.
+            att_pos < intel_pos
+        }
+        (Some(_intel), None) => false,
+        (None, Some(_att)) => true,
+        (None, None) => {
+            match () {
+                _ if asm.contains("word ptr") => false,
+                // Guess based on sigils used in AT&T assembly:
+                // $ for constants, % for registers, and ( for address calculations
+                _ if asm.contains('$') || asm.contains('%') || asm.contains('(') => true,
+                _ if asm.contains('[') => false,
+                // default to true, because AT&T is the default for gcc inline asm
+                _ => true,
+            }
```

Comment:
This seems a lot more readable though because you get one `if`, all containing an `asm.contains()` on each line.  And if you use `if`, `else if`, ..., `else`, then `clippy` complains about the final `else { true }` and wants to turn that into a `!asm.contains('[')`, which throws off the consistency with the other branches.  That's the warning I was trying to avoid while still keeping the consistency.

Reply:
Readability is a matter of communication--getting the intended semantics into the reader's head. Matching on `()` is contrary to this goal, because it introduces entities (`()`, the wildcard patterns, and the match itself) that have nothing to do with the goal of examining `asm`. These serve only to misdirect the reader until they see the rest of what's going on.

I was proposing:
```rust
if asm.contains("word ptr") { return false }
// Guess based on sigils used in AT&T assembly:
// $ for constants, % for registers, and ( for address calculations
if asm.contains('$') || asm.contains('%') || asm.contains('(') { return true }
if asm.contains('[') { return false }
// default to true, because AT&T is the default for gcc inline asm
true
```

Reply:
Yeah, but that would be fine, but it goes against both `rustfmt` or `clippy`, which I think we should follow.

The version this way that passes `rustfmt` and `clippy` would be:

```rust
            if asm.contains("word ptr") {
                return false;
            }
            // Guess based on sigils used in AT&T assembly:
            // $ for constants, % for registers, and ( for address calculations
            if asm.contains('$') || asm.contains('%') || asm.contains('(') {
                return true;
            }
            // default to true, because AT&T is the default for gcc inline asm
            !asm.contains('[')
```

which throws off the consistency of the branches with the `!asm.contains('[')`, and the comment makes less sense now, too.

Reply:
I think clippy is just wrong here--returning true as an explicit default is not inherently a mistake.

Reply:
Sounds perfect.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/462/commits/87c88956a80c92c6711956b4c15d64718caf46c6.